src/contracts/NiftyupFactory.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+
+import '@openzeppelin/contracts/proxy/Clones.sol';
+import "./NiftyupNFT.sol";
+
+
+contract NiftyupFactory {
+  NiftyupNFT immutable public implementation;
+
+  event CreatedNFT(address indexed creator, address indexed owner, address indexed nft);
```

Comment:
what do you think of renaming to `CreatedNFTContract` ? so its more clear its for a new contract and not confused with a mint of an item in the contract